app/main.py

```diff
@@ -1,25 +1,91 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: str, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: callable, owner: callable) -> None:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: callable, value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
+        else:
+            setattr(
+                instance,
+                self.protected_name,
+                f"Need {self.public_name} in range from {self.min_amount} "
+                f"to {self.max_amount}"
+            )
```

Comment:
You should check that value is an integer and that value is in the correct range, if not -> raise ValueError